inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp

```diff
@@ -1508,39 +1849,70 @@ void MKLDNNReduceNode::initSupportedPrimitiveDescriptors() {
     }
 }
 
+void MKLDNNReduceNode::prepareParams() {
+    src_dims = getParentEdgesAtPort(REDUCE_DATA)[0]->getMemory().getDesc().getShape().getDims();
+    reduce_axes = raw_axes;
+    if (jit_mode && jit_beyond_5D) {
+        update_src_dims();
+    }
+
+    dst_size = getChildEdgeAt(0)->getMemoryPtr()->GetSize();
+    calc_process_dst_dims();
+    if (jit_mode) {
+        set_reduce_dim_flags();
+    }
+}
+
 void MKLDNNReduceNode::createPrimitive() {
     auto &dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
-    auto &srcDataMemPtr = getParentEdgeAt(REDUCE_DATA)->getMemoryPtr();
-    auto &srcIndexesMemPtr = getParentEdgeAt(REDUCE_INDEXES)->getMemoryPtr();
+    auto &srcMemPtr = getParentEdgeAt(REDUCE_DATA)->getMemoryPtr();
     if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
         IE_THROW() << errorPrefix << " has not allocated destination memory.";
-    if (!srcDataMemPtr || !srcDataMemPtr->GetPrimitivePtr() || !srcIndexesMemPtr || !srcIndexesMemPtr->GetPrimitivePtr())
+    if (!srcMemPtr || !srcMemPtr->GetPrimitivePtr())
         IE_THROW() << errorPrefix << " has not allocate input memory.";
     if (getSelectedPrimitiveDescriptor() == nullptr)
         IE_THROW() << errorPrefix << " has nullable preferable primitive descriptor";
 
-    auto selectedPD = getSelectedPrimitiveDescriptor();
-    planar_layout = getParentEdgeAt(REDUCE_DATA)->getMemory().getDesc().hasLayoutType(LayoutType::ncsp);
+    if (getParentEdgeAt(REDUCE_DATA)->getMemory().getDesc().hasLayoutType(LayoutType::ncsp)) {
```

Comment:
It is better to use `srcMemPtr` as it is under your hand already, instead of calling `getParentEdgeAt(REDUCE_DATA)->getMemory()` again.

Reply:
Fixed. Thanks Maksim!